---
title: GSE96058 R Data Package
always_allow_html: yes
output: github_document   ### will keep md file!!!
bibliography: docs/_bibFiles/bibFile.bib
csl: docs/_bibFiles/cell-numeric.csl
link-citations: true
---

---

<!--
[![Travis build status](https://travis-ci.com/12379Monty/GSE96058.svg?branch=master)](https://travis-ci.com/12379Monty/GSE96058)
-->

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r global-options, include = FALSE}

knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)

 options(knitr.table.format = 'html')

```


This package contains data from NCBI GEO 
[Series GSE96058](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE96058).
3,678 patients with breast cancer (BC) were studied.  For 405 tumors, a comprehensive 
multi-rater histopathologic evaluation was performed. Using RNA-seq data, 
single-gene classifiers and multigene classifiers (MGCs) were trained on consensus 
histopathology labels. Trained classifiers were tested on a prospective 
population-based series of 3,273 BCs that included a median follow-up of 52 
months (Sweden Cancerome Analysis Networkâ€”Breast [SCAN-B], 
ClinicalTrials.gov identifier: NCT02306096), and results were evaluated by 
agreement statistics and Kaplan-Meier and Cox survival analyses. 
See  Brueffer et al. (2018) [@Brueffer:2018aa] for a detailed report
of the findings.

### User notes

* The data in NCBI GEO
[Series GSE96058](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE96058)
does not contain any of the training samples discussed in the Brueffer et al. (2018) [@Brueffer:2018aa]
paper.  This package therefore only contains samples from the prospective
population-based series of 3,273 BCs.

* The gene expression data are stored in several tables due to github file size
restrictions:
    - featureCount_repl1, featureCount_repl2: 30865 x 136 tables of 
counts for replicated samples
    - featureCount_sub1-_sub5: 30865 x 626-629 tables of counts 
for random subsets of samples, balanced over pam scores.

*  We use `featureCount` although the gene coverage measures
stored in the GEO data set are `FPKM` estimates of gene expression.

<br/>

## Installation

``` r
# install.packages("devtools")
devtools::install_github("12379Monty/GSE96058")
```


## Data sets  

```{r dataSets, eval=T, echo=F}

library(magrittr)

tmp <- data(package='GSE96058')
knitr::kable(tmp$results[, c("Item", "Title")]) %>%
  kableExtra::kable_styling(full_width = F)


```


## Data Processing

The data processing step descriptions were extracted from the sample 
table in the GSE96058 dataset and stored as a separate data from
in the GSE96058 data package; `data_processing`

```{r data-processing, eval=T}

library(GSE96058)

knitr::kable(data_processing,
  row.names=F)  %>%
  kableExtra::kable_styling(full_width = F)

```



## Examples

### ER/PR/HER2 status

* To  identify breast cancer associated read-through fusion transcripts,
Varley et al. (2014) [@Varley:2014aa] analyzed the paired-end
RNA-seq data from of 168 breast samples, including 28 breast cancer
cell lines, 42 triple negative breast cancer primary tumors,
42 estrogen receptor positive breast cancer primary
tumors, and 56 non-malignant breast tissue samples. Data
are deposited in [GEO GSE58135](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE58135)

* Analyzing this dataset Li et al. (2018) [@Li:2018aa] identified 
797 DEGs uniquely expressed in triple negative BC (TNBC) and 1403 DEGs uniquely expressed 
in estrogen positive and HER2 negative BC (ER+HER2-BC).   

This dataset has a limited number of TNBC and ER+HER2-BC that could be examined
to verify the findings in Li et al. (2018) [@Li:2018aa]. 

```{r, eval=T}

library(magrittr)
library(GSE96058)

ER_PR_HER2_tbl <- with(sampDesc %>% 
 dplyr::filter(!isRepl & er_Status!='NA'  & pgr_Status!='NA' & her2_Status!='NA'),
 table(ER_PR_HER2=paste(er_Status, pgr_Status, her2_Status, sep='_'),
  exclude=NULL)
 )

 knitr::kable(ER_PR_HER2_tbl,
  caption="ER, PR and HER2 Status")  %>%
  kableExtra::kable_styling(full_width = F)


```

<br/>

###  ER status vs predicted value

* The GSE58135 dataset contains single gene and multi gene predictors
of ER/PR/HER2/ki67 status:    
        -  `r setdiff(grep('_Status', names(sampDesc), value=T),  'lymphNode_Status')`    
        -  `r grep('SGC', names(sampDesc), value=T)`   
        -  `r grep('MGC', names(sampDesc), value=T)`  


```{r er-status-pred}

SGC_tbl <- with(sampDesc,
 table(er_Status, er_SGC)
)
dimnames(SGC_tbl)[[1]] <- paste0('ER=', dimnames(SGC_tbl)[[1]])

MGC_tbl <- with(sampDesc,
 table(er_Status, er_MGC)
)
dimnames(MGC_tbl)[[1]] <- paste0('ER=', dimnames(MGC_tbl)[[1]])


knitr::kable(SGC_tbl, caption='ER Status - SGC Predictions')  %>%
  kableExtra::kable_styling(full_width = F)

knitr::kable(MGC_tbl, caption='ER Status - MGC Predictions')  %>%
  kableExtra::kable_styling(full_width = F)

```

Similarly for PGR, HER2 and ki67.


<br/>



### Treatment and survival.

* The GSE58135 dataset contains treatment and survival data


```{r chemo-survival, fig.height=4, fig.width=6}

chemo_surv_lst <- with(sampDesc %>% dplyr::filter(chemo_treated %in% c('0','1')),
  split(ovrallSurvDays, 
   paste0('chemo_',chemo_treated, '\nSrvEv_', ovrallSurvEvent))
)

boxplot(chemo_surv_lst, ylab='ovrallSurvDays')
title("Survival Days vs chemo_treated*ovrallSurvEvent")

```

Can similarly examine survival vs endocrine_treated.

```{r endocrine-survival, fig.height=4, fig.width=6}

endocrine_surv_lst <- with(sampDesc %>% dplyr::filter(endocrine_treated %in% c('0','1')),
  split(ovrallSurvDays, 
   paste0('endocrine_',endocrine_treated, '\nSrvEv_', ovrallSurvEvent))
)

boxplot(endocrine_surv_lst, ylab='ovrallSurvDays')
title("Survival Days vs endocrine_treated*ovrallSurvEvent")

```

<br/>


###  PAM50 subtyping 

* GSE96058 has PAM50 subtypung information - see data processing above.
Samples were distributed across different count tables in order to meet
github file size constraints.  This allocation was done to keep
the PAM50 score distribution constant. 


```{r look-pam50, fig.height=4, fig.width=6}

pam50_byTable_tbl <- with(sampDesc,
 table(pam50_subtype, countTable))

barplot(pam50_byTable_tbl, beside=T, legend=T, args.legend=list(x='topleft'))
title("Sample PAM subtype by subset")


```

Let's examine how counts cluster for one of the subsets.

First look at coverage, ordered by sample ID (sometimes a proxy for
processing time effects)

```{r bxp-repl1, fig.height=4, fig.width=10}
KellyColors.vec <- c(
  "#222222", "#F3C300", "#875692", "#F38400", "#A1CAF1",
  "#BE0032", "#C2B280", "#848482", "#008856", "#E68FAC", "#0067A5",
  "#F99379", "#604E97", "#F6A600", "#B3446C", "#DCD300", "#882D17",
  "#8DB600", "#654522", "#E25822", "#2B3D26"
)
col_vec <- KellyColors.vec

pamVal_vec <- with(sampDesc, unique(pam50_subtype))
pamCol_vec <- col_vec[1:length(pamVal_vec)]
names(pamCol_vec) <- pamVal_vec

sampDesc_repl1 <- sampDesc[match(colnames(featureCount_repl1), sampDesc$title),]

o.v <- order(sampDesc_repl1$bioSample)

sampDesc_repl1 <- sampDesc_repl1[o.v,]
featureCount_repl1 <- featureCount_repl1[, o.v]
rm(o.v)

par(mfrow = c(1, 1), mar = c(2, 2, 2, 4), oma = c(0, 1, 4, 2))

# AqNPATisMix outline=T for LBO
boxplot(featureCount_repl1,
  add = F,
  ylim = c(-4, 4), 
  staplewex = 0, # remove horizontal whisker lines
  staplecol = "white", # just to be totally sure
  outline = F, # remove outlying points
  whisklty = 0, # remove vertical whisker lines
  las = 2, horizontal = F, xaxt = "n",
  border = pamCol_vec[sampDesc_repl1$pam50_subtype])
title("Coverage for repl1 samples ordered by bioSample ID")

legend('top', text.col=pamCol_vec, legend=names(pamCol_vec), bty='n', horiz=T)


```

```{r density-repl1, fig.height=4, fig.width=10}

par(mfcol = c(1, 1), mar = c(2, 2, 1, 1), oma = c(0, 3, 2, 0))


plot(density(featureCount_repl1[, 1]),
  col = pamCol_vec[sampDesc_repl1$pam50_subtype[1]],
  lty = 1, lwd = 2, 
  xlim = c(-4, 7), ylim = c(0, .4), las = 2, main = "", xlab = ""
)
title("Coverage densities for repl1 samples")

for (JJ in 2:ncol(featureCount_repl1)) {
  den <- density(featureCount_repl1[, JJ])
  lines(den$x, den$y,
    col = pamCol_vec[sampDesc_repl1$pam50_subtype[JJ]],
    lty = 1)
}

legend('top', text.col=pamCol_vec, legend=names(pamCol_vec), bty='n', horiz=T)


```

FPKM of zero looks like a good place to separate expressed from weakly expressed
genes.  Apply this filter, requiring at a gene to be epxressed beyond this 
nominal value in at least 10 samples to be considered for this analysis.

```{r bxp-repl1-filtered, fig.height=4, fig.width=10}

weak.flg <- rowSums(featureCount_repl1 > 0)  < 10

cat("Excluding", round(100*mean(weak.flg),1), "percent of genes.\n")
cat(sum(!weak.flg), 'genes kept for analysis.\n')

par(mfrow = c(1, 1), mar = c(2, 2, 2, 4), oma = c(0, 1, 4, 2))

# AqNPATisMix outline=T for LBO
boxplot(featureCount_repl1[!weak.flg,],
  add = F,
  ylim = c(-1, 5), 
  staplewex = 0, # remove horizontal whisker lines
  staplecol = "white", # just to be totally sure
  outline = F, # remove outlying points
  whisklty = 0, # remove vertical whisker lines
  las = 2, horizontal = F, xaxt = "n",
  border = pamCol_vec[sampDesc_repl1$pam50_subtype])
title("Filtered Coverage for repl1 samples ordered by bioSample ID")

legend('top', text.col=pamCol_vec, legend=names(pamCol_vec), bty='n', horiz=T)


```






```{r plotMDS-repl1, cache=T, fig.height=4, fig.width=10}

par(mfcol = c(1, 2), mar = c(4, 4, 2, 1), xpd = NA, oma = c(0, 0, 2, 0))

MDS.out <- limma::plotMDS(featureCount_repl1[!weak.flg,],
  col = pamCol_vec[sampDesc_repl1$pam50_subtype],
  pch=1
)


MDS.out <- limma::plotMDS(featureCount_repl1[!weak.flg,],
  col = pamCol_vec[sampDesc_repl1$pam50_subtype],
  pch=1,
  dim.plot = 3:4
)

mtext(outer=T, side=3,
 "MDS plots filtered coverage in repl1 samples - Col = pam50_subtype")

```

We see some association with pam subtype, but no strong clustering
in this display.  


## Package page

* https://12379monty.github.io/GSE96058/

<!-- DOESNT WORK WITH GITHUB
## References
<div id="refs"></div>
-->

-----

Please note that the 'GSE96058' project is released with a [Contributor Code of Conduct](.github/CODE_OF_CONDUCT.md). By contributing to this project, you agree to abide by its terms.

----- 

## References

<div id="refs"></div>

<!--

Rscript -e "rmarkdown::render('README.Rmd')"

# nohup Rscript -e "rmarkdown::render('README.Rmd')" > README.log  &


-->
