<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Data for NCBI GEO Series GSE96058 • GSE96058</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Data for NCBI GEO Series GSE96058">
<meta property="og:description" content="Data for NCBI GEO Series GSE112679 are organized
    into a set of R objects convenient for analyis.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">GSE96058</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="gse96058-r-data-package" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#gse96058-r-data-package" class="anchor"></a>GSE96058 R Data Package</h1></div>
<hr>
<!--
[![Travis build status](https://travis-ci.com/12379Monty/GSE96058.svg?branch=master)](https://travis-ci.com/12379Monty/GSE96058)
--><!-- README.md is generated from README.Rmd. Please edit that file --><p>This package contains data from NCBI GEO <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE96058">Series GSE96058</a>. 3,678 patients with breast cancer (BC) were studied. For 405 tumors, a comprehensive multi-rater histopathologic evaluation was performed. Using RNA-seq data, single-gene classifiers and multigene classifiers (MGCs) were trained on consensus histopathology labels. Trained classifiers were tested on a prospective population-based series of 3,273 BCs that included a median follow-up of 52 months (Sweden Cancerome Analysis Network—Breast <br><span class="math display"><em>S</em><em>C</em><em>A</em><em>N</em> − <em>B</em></span><br>, ClinicalTrials.gov identifier: NCT02306096), and results were evaluated by agreement statistics and Kaplan-Meier and Cox survival analyses. See Brueffer et al. (2018) <br><span class="math display">$$[1](#ref-Brueffer:2018aa)$$</span><br> for a detailed report of the findings.</p>
<div id="user-notes" class="section level3">
<h3 class="hasAnchor">
<a href="#user-notes" class="anchor"></a>User notes</h3>
<ul>
<li><p>The data in NCBI GEO <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE96058">Series GSE96058</a> does not contain any of the training samples discussed in the Brueffer et al. (2018) <br><span class="math display">$$[1](#ref-Brueffer:2018aa)$$</span><br> paper. This package therefore only contains samples from the prospective population-based series of 3,273 BCs.</p></li>
<li>
<p>The gene expression data are stored in several tables due to github file size restrictions:</p>
<ul>
<li>featureCount_repl1, featureCount_repl2: 30865 x 136 tables of counts for replicated samples</li>
<li>featureCount_sub1-_sub5: 30865 x 626-629 tables of counts for random subsets of samples, balanced over pam scores.</li>
</ul>
</li>
<li><p>We use <code>featureCount</code> although the gene coverage measures stored in the GEO data set are <code>FPKM</code> estimates of gene expression.</p></li>
</ul>
<p><br></p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="co"># install.packages("devtools")</span>
<span class="kw">devtools</span>::<span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span>(<span class="st">"12379Monty/GSE96058"</span>)
</pre></div>
</div>
<div id="data-sets" class="section level2">
<h2 class="hasAnchor">
<a href="#data-sets" class="anchor"></a>Data sets</h2>
<table class="table table">
<thead><tr>
<th style="text-align:left;">
<p>Item</p>
</th>
<th style="text-align:left;">
<p>Title</p>
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
<p>data_processing</p>
</td>
<td style="text-align:left;">
<p>A data frame with one column describing data processing in GSE96058</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>featureCount_repl1</p>
</td>
<td style="text-align:left;">
<p>A matrix of feature counts for replicated samples - replicate 1</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>featureCount_repl2</p>
</td>
<td style="text-align:left;">
<p>A matrix of feature counts for replicated samples - replicate 2</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>featureCount_sub1</p>
</td>
<td style="text-align:left;">
<p>A matrix of feature counts for a random subset of samples balanced according to pam50 category</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>featureCount_sub2</p>
</td>
<td style="text-align:left;">
<p>A matrix of feature counts for a random subset of samples balanced according to pam50 category</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>featureCount_sub3</p>
</td>
<td style="text-align:left;">
<p>A matrix of feature counts for a random subset of samples balanced according to pam50 category</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>featureCount_sub4</p>
</td>
<td style="text-align:left;">
<p>A matrix of feature counts for a random subset of samples balanced according to pam50 category</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>featureCount_sub5</p>
</td>
<td style="text-align:left;">
<p>A matrix of feature counts for a random subset of samples balanced according to pam50 category</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>genes_annot</p>
</td>
<td style="text-align:left;">
<p>A data frame describing the features</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>sampDesc</p>
</td>
<td style="text-align:left;">
<p>A data frame describing the samples in the GSE96058 dataset</p>
</td>
</tr>
</tbody>
</table>
</div>
<div id="data-processing" class="section level2">
<h2 class="hasAnchor">
<a href="#data-processing" class="anchor"></a>Data Processing</h2>
<p>The data processing step descriptions were extracted from the sample table in the GSE96058 dataset and stored as a separate data from in the GSE96058 data package; <code>data_processing</code></p>
<div class="sourceCode" id="cb2"><pre class="downlit">

<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw">GSE96058</span>)

<span class="kw">knitr</span>::<span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="kw">data_processing</span>,
  row.names=<span class="kw">F</span>)  <span class="op">%&gt;%</span>
  <span class="kw">kableExtra</span>::<span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_styling.html">kable_styling</a></span>(full_width = <span class="kw">F</span>)
</pre></div>
<table class="table table">
<thead><tr>
<th style="text-align:left;">
<p>GSE96058_Data_Processing</p>
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
<p>Base-calling using manufacturer’s on-instrument software.</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>Demultiplexing was done with Picard versions 1.120 or 1.128. IlluminaBasecallsToFastq parameters used were ADAPTERS_TO_CHECK=INDEXED, ADAPTERS_TO_CHECK=PAIRED_END, INCLUDE_NON_PF_READS=false</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>Filtering to remove reads that align (using Bowtie 2 with default parameters except -k 1 –phred33 –local) to ribosomal RNA/DNA (GenBank loci NR_023363.1, NR_003285.2, NR_003286.2, NR_003287.2, X12811.1, U13369.1), phiX174 Illumina control (NC_001422.1), and sequences contained in the UCSC RepeatMasker track (downloaded March 14, 2011).</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>Fragment size distribution (mean and width) for the alignment step was estimated for each sample using bowtie2 2.2.3 and 2.2.5. Parameter set during estimation were -fr, -k 1, –phred33, –local, and -u 100000, using human genome assembly GRCh38.</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>Remaining reads were aligned using TopHat2 2.0.12 or 2.0.13 (default parameters except for –mate-inner-dist X (estimated in previous step), –mate-std-dev Y (estimated in previous step), –library-type fr-firststrand, –no-coverage-search, –max-insertion-length 20, –max-deletion-length 20, –read-gap-length 20, –read-edit-dist 22) to the human genome reference GRCh38 together with 104,133 transcript annotations from the UCSC knownGenes table (downloaded September 22, 2014).</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>Gene expression data in FPKM were generated using cufflinks 2.2.1 (default parameters except –GTF, –frag-bias-correct GRCh38.fa, –multi-read-correct, –library-type fr-firststrand, –total-hits-norm, –max-bundle-frags 10000000). The resulting data was was post-processed by collapsing on 30,865 unique gene symbols (sum of FPKM values of each matching transcript), adding to each expression measurement 0.1 FPKM, and performing a log2 transformation.</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>PAM50 subtyping was performed using an implementation of the Parker method (Parker et al., J.Clin Oncol 2009). In short, to avoid context dependency when assigning PAM50 subtype by nearest-centroid, a fixed reference was selected to match the original cohort used by Parker et al. with respect to available clinical characteristics. Before subtyping tumors in this study, gene expression of the PAM50 genes for each tumor was centered to the reference set separately using custom R scripts.</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>Single-gene classifiers (SGCs) and multi-gene classifiers (MGCs) were trained for ER, PgR, HER2, Ki67 (SGC and MGC) and NHG (MGC only) on the expression of the single underlying gene (ESR1, PGR, ERBB2 or MKI67) or multiple genes (5000 most varying genes across all samples) of a 405 sample cohort using consensus pathology scores as labels. Classifier training was performed by selecting expression thresholds that maximize prediction concordance to the consensus scores (SGCs) and nearest shrunken centroids using the pamr R package (MGCs). The classifiers were used to predict the biomarker status in a validation cohort of 3273 samples from patients enrolled in the SCAN-B study.</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>Genome_build: Human genome reference GRCh38/hg38.</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>Supplementary_files_format_and_content: Gene expression in FPKM in CSV format.</p>
</td>
</tr>
</tbody>
</table>
</div>
<div id="examples" class="section level2">
<h2 class="hasAnchor">
<a href="#examples" class="anchor"></a>Examples</h2>
<div id="erprher2-status" class="section level3">
<h3 class="hasAnchor">
<a href="#erprher2-status" class="anchor"></a>ER/PR/HER2 status</h3>
<ul>
<li><p>To identify breast cancer associated read-through fusion transcripts, Varley et al. (2014) <br><span class="math display">$$[2](#ref-Varley:2014aa)$$</span><br> analyzed the paired-end RNA-seq data from of 168 breast samples, including 28 breast cancer cell lines, 42 triple negative breast cancer primary tumors, 42 estrogen receptor positive breast cancer primary tumors, and 56 non-malignant breast tissue samples. Data are deposited in <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE58135">GEO GSE58135</a></p></li>
<li><p>Analyzing this dataset Li et al. (2018) <br><span class="math display">$$[3](#ref-Li:2018aa)$$</span><br> identified 797 DEGs uniquely expressed in triple negative BC (TNBC) and 1403 DEGs uniquely expressed in estrogen positive and HER2 negative BC (ER+HER2-BC).</p></li>
</ul>
<p>This dataset has a limited number of TNBC and ER+HER2-BC that could be examined to verify the findings in Li et al. (2018) <br><span class="math display">$$[3](#ref-Li:2018aa)$$</span><br>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit">

<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw">magrittr</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw">GSE96058</span>)

<span class="kw">ER_PR_HER2_tbl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span>(<span class="kw">sampDesc</span> <span class="op">%&gt;%</span> <span class="kw">dplyr</span>::<span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="op">!</span><span class="kw">isRepl</span>), 
 <span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(ER_PR_HER2=<span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="kw">er_Status</span>, <span class="kw">pgr_Status</span>, <span class="kw">her2_Status</span>, sep=<span class="st">'_'</span>),
  exclude=<span class="kw">NULL</span>)
 )))

 <span class="kw">knitr</span>::<span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span>(<span class="kw">ER_PR_HER2_tbl</span>)),
  caption=<span class="st">"ER, PR and HER2 Status"</span>)  <span class="op">%&gt;%</span>
  <span class="kw">kableExtra</span>::<span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_styling.html">kable_styling</a></span>(full_width = <span class="kw">F</span>)
</pre></div>
<table class="table table">
<caption>
<p>ER, PR and HER2 Status</p>
</caption>
<tbody>
<tr>
<td style="text-align:left;">
<p>0_0_0</p>
</td>
<td style="text-align:right;">
<p>143</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>0_0_1</p>
</td>
<td style="text-align:right;">
<p>56</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>0_0_NA</p>
</td>
<td style="text-align:right;">
<p>12</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>0_1_0</p>
</td>
<td style="text-align:right;">
<p>22</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>0_1_1</p>
</td>
<td style="text-align:right;">
<p>6</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>0_1_NA</p>
</td>
<td style="text-align:right;">
<p>1</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>0_NA_1</p>
</td>
<td style="text-align:right;">
<p>1</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>1_0_0</p>
</td>
<td style="text-align:right;">
<p>126</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>1_0_1</p>
</td>
<td style="text-align:right;">
<p>44</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>1_0_NA</p>
</td>
<td style="text-align:right;">
<p>5</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>1_1_0</p>
</td>
<td style="text-align:right;">
<p>2198</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>1_1_1</p>
</td>
<td style="text-align:right;">
<p>232</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>1_1_NA</p>
</td>
<td style="text-align:right;">
<p>90</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>1_NA_0</p>
</td>
<td style="text-align:right;">
<p>101</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>1_NA_1</p>
</td>
<td style="text-align:right;">
<p>34</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>1_NA_NA</p>
</td>
<td style="text-align:right;">
<p>2</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>NA_1_0</p>
</td>
<td style="text-align:right;">
<p>2</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>NA_1_1</p>
</td>
<td style="text-align:right;">
<p>3</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>NA_NA_0</p>
</td>
<td style="text-align:right;">
<p>139</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>NA_NA_1</p>
</td>
<td style="text-align:right;">
<p>44</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>NA_NA_NA</p>
</td>
<td style="text-align:right;">
<p>12</p>
</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
<div id="er-status-vs-predicted-value" class="section level3">
<h3 class="hasAnchor">
<a href="#er-status-vs-predicted-value" class="anchor"></a>ER status vs predicted value</h3>
<ul>
<li>The GSE58135 dataset contains single gene and multi gene predictors of ER/PR/HER2/ki67 status:<br>
- er_Status, her2_Status, ki67_Status, pgr_Status<br>
- er_SGC, her2_SGC, ki67_SGC, pgr_SGC<br>
- er_MGC, her2_MGC, ki67_MGC, nhg_MGC, pgr_MGC</li>
</ul>
<!-- end list --><div class="sourceCode" id="cb4"><pre class="downlit">

<span class="kw">SGC_tbl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span>(<span class="kw">sampDesc</span>,
 <span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="kw">er_Status</span>, <span class="kw">er_SGC</span>)
)
<span class="fu"><a href="https://rdrr.io/r/base/dimnames.html">dimnames</a></span>(<span class="kw">SGC_tbl</span>)[[<span class="fl">1</span>]] <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">'ER='</span>, <span class="fu"><a href="https://rdrr.io/r/base/dimnames.html">dimnames</a></span>(<span class="kw">SGC_tbl</span>)[[<span class="fl">1</span>]])

<span class="kw">MGC_tbl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span>(<span class="kw">sampDesc</span>,
 <span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="kw">er_Status</span>, <span class="kw">er_MGC</span>)
)
<span class="fu"><a href="https://rdrr.io/r/base/dimnames.html">dimnames</a></span>(<span class="kw">MGC_tbl</span>)[[<span class="fl">1</span>]] <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">'ER='</span>, <span class="fu"><a href="https://rdrr.io/r/base/dimnames.html">dimnames</a></span>(<span class="kw">MGC_tbl</span>)[[<span class="fl">1</span>]])


<span class="kw">knitr</span>::<span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="kw">SGC_tbl</span>, caption=<span class="st">'ER Status - SGC Predictions'</span>)  <span class="op">%&gt;%</span>
  <span class="kw">kableExtra</span>::<span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_styling.html">kable_styling</a></span>(full_width = <span class="kw">F</span>)
</pre></div>
<table class="table table">
<caption>
<p>ER Status - SGC Predictions</p>
</caption>
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
<p>0</p>
</th>
<th style="text-align:right;">
<p>1</p>
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
<p>ER=0</p>
</td>
<td style="text-align:right;">
<p>192</p>
</td>
<td style="text-align:right;">
<p>62</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>ER=1</p>
</td>
<td style="text-align:right;">
<p>65</p>
</td>
<td style="text-align:right;">
<p>2870</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>ER=NA</p>
</td>
<td style="text-align:right;">
<p>137</p>
</td>
<td style="text-align:right;">
<p>83</p>
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb5"><pre class="downlit">

<span class="kw">knitr</span>::<span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="kw">MGC_tbl</span>, caption=<span class="st">'ER Status - MGC Predictions'</span>)  <span class="op">%&gt;%</span>
  <span class="kw">kableExtra</span>::<span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_styling.html">kable_styling</a></span>(full_width = <span class="kw">F</span>)
</pre></div>
<table class="table table">
<caption>
<p>ER Status - MGC Predictions</p>
</caption>
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
<p>0</p>
</th>
<th style="text-align:right;">
<p>1</p>
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
<p>ER=0</p>
</td>
<td style="text-align:right;">
<p>242</p>
</td>
<td style="text-align:right;">
<p>12</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>ER=1</p>
</td>
<td style="text-align:right;">
<p>246</p>
</td>
<td style="text-align:right;">
<p>2689</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>ER=NA</p>
</td>
<td style="text-align:right;">
<p>205</p>
</td>
<td style="text-align:right;">
<p>15</p>
</td>
</tr>
</tbody>
</table>
<p>Similarly for PGR, HER2 and ki67.</p>
<p><br></p>
</div>
<div id="treatment-and-survival" class="section level3">
<h3 class="hasAnchor">
<a href="#treatment-and-survival" class="anchor"></a>Treatment and survival.</h3>
<ul>
<li>The GSE58135 dataset contains treatment and survival data</li>
</ul>
<!-- end list --><div class="sourceCode" id="cb6"><pre class="downlit">
 <span class="co">###, fig.cap="Survival Days vs chemo_treated*ovrallSurvEvent"}</span>

<span class="kw">chemo_surv_lst</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span>(<span class="kw">sampDesc</span> <span class="op">%&gt;%</span> <span class="kw">dplyr</span>::<span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="kw">chemo_treated</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'0'</span>,<span class="st">'1'</span>)),
  <span class="fu"><a href="https://rdrr.io/r/base/split.html">split</a></span>(<span class="kw">ovrallSurvDays</span>, 
   <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">'chemo_'</span>,<span class="kw">chemo_treated</span>, <span class="st">'\nSrvEv_'</span>, <span class="kw">ovrallSurvEvent</span>))
)

<span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html">boxplot</a></span>(<span class="kw">chemo_surv_lst</span>, ylab=<span class="st">'ovrallSurvDays'</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/title.html">title</a></span>(<span class="st">"Survival Days vs chemo_treated*ovrallSurvEvent"</span>)
</pre></div>
<p><img src="reference/figures/README-chemo-survival-1.png" width="100%"></p>
<p>Can similarly examine survival vs endocrine_treated.</p>
<div class="sourceCode" id="cb7"><pre class="downlit">
 <span class="co">###, fig.cap="Survival Days vs endocrine_treated*ovrallSurvEvent",eval=F, echo=F}</span>

<span class="kw">endocrine_surv_lst</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span>(<span class="kw">sampDesc</span> <span class="op">%&gt;%</span> <span class="kw">dplyr</span>::<span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="kw">endocrine_treated</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'0'</span>,<span class="st">'1'</span>)),
  <span class="fu"><a href="https://rdrr.io/r/base/split.html">split</a></span>(<span class="kw">ovrallSurvDays</span>, 
   <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">'endocrine_'</span>,<span class="kw">endocrine_treated</span>, <span class="st">'\nSrvEv_'</span>, <span class="kw">ovrallSurvEvent</span>))
)

<span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html">boxplot</a></span>(<span class="kw">endocrine_surv_lst</span>, ylab=<span class="st">'ovrallSurvDays'</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/title.html">title</a></span>(<span class="st">"Survival Days vs endocrine_treated*ovrallSurvEvent"</span>)
</pre></div>
<p><img src="reference/figures/README-endocrine-survival-1.png" width="100%"></p>
<p><br></p>
</div>
<div id="pam50-subtyping" class="section level3">
<h3 class="hasAnchor">
<a href="#pam50-subtyping" class="anchor"></a>PAM50 subtyping</h3>
<ul>
<li>GSE96058 has PAM50 subtypung information - see data processing above. Samples were distributed across different count tables in order to meet github file size constraints. This allocation was done to keep the PAM50 score distribution constant.</li>
</ul>
<!-- end list --><div class="sourceCode" id="cb8"><pre class="downlit">
 <span class="co">###, fig.cap="Sapmple PAM subtype vs Count Table" }</span>

<span class="kw">pam50_byTable_tbl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span>(<span class="kw">sampDesc</span>,
 <span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="kw">pam50_subtype</span>, <span class="kw">countTable</span>))

<span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html">barplot</a></span>(<span class="kw">pam50_byTable_tbl</span>, beside=<span class="kw">T</span>, legend=<span class="kw">T</span>, args.legend=<span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(x=<span class="st">'topleft'</span>))
<span class="fu"><a href="https://rdrr.io/r/graphics/title.html">title</a></span>(<span class="st">"Sample PAM subtype by subset"</span>)
</pre></div>
<p><img src="reference/figures/README-look-pam50-1.png" width="100%"></p>
<p>Let’s examine how counts cluster for one of the subsets.</p>
<p>First look at coverage, ordered by sample ID (sometimes a proxy for processing time effects)</p>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="co">###, fig.cap="Coverage for repl1 samples ordered by bioSample ID"}</span>
<span class="kw">KellyColors.vec</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(
  <span class="st">"#222222"</span>, <span class="st">"#F3C300"</span>, <span class="st">"#875692"</span>, <span class="st">"#F38400"</span>, <span class="st">"#A1CAF1"</span>,
  <span class="st">"#BE0032"</span>, <span class="st">"#C2B280"</span>, <span class="st">"#848482"</span>, <span class="st">"#008856"</span>, <span class="st">"#E68FAC"</span>, <span class="st">"#0067A5"</span>,
  <span class="st">"#F99379"</span>, <span class="st">"#604E97"</span>, <span class="st">"#F6A600"</span>, <span class="st">"#B3446C"</span>, <span class="st">"#DCD300"</span>, <span class="st">"#882D17"</span>,
  <span class="st">"#8DB600"</span>, <span class="st">"#654522"</span>, <span class="st">"#E25822"</span>, <span class="st">"#2B3D26"</span>
)
<span class="kw">col_vec</span> <span class="op">&lt;-</span> <span class="kw">KellyColors.vec</span>

<span class="kw">pamVal_vec</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span>(<span class="kw">sampDesc</span>, <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="kw">pam50_subtype</span>))
<span class="kw">pamCol_vec</span> <span class="op">&lt;-</span> <span class="kw">col_vec</span>[<span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="kw">pamVal_vec</span>)]
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="kw">pamCol_vec</span>) <span class="op">&lt;-</span> <span class="kw">pamVal_vec</span>

<span class="kw">sampDesc_repl1</span> <span class="op">&lt;-</span> <span class="kw">sampDesc</span>[<span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="kw">featureCount_repl1</span>), <span class="kw">sampDesc</span><span class="op">$</span><span class="kw">title</span>),]

<span class="kw">o.v</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span>(<span class="kw">sampDesc_repl1</span><span class="op">$</span><span class="kw">bioSample</span>)

<span class="kw">sampDesc_repl1</span> <span class="op">&lt;-</span> <span class="kw">sampDesc_repl1</span>[<span class="kw">o.v</span>,]
<span class="kw">featureCount_repl1</span> <span class="op">&lt;-</span> <span class="kw">featureCount_repl1</span>[, <span class="kw">o.v</span>]
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span>(<span class="kw">o.v</span>)

<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(mfrow = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>, <span class="fl">1</span>), mar = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">4</span>), oma = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">4</span>, <span class="fl">2</span>))

<span class="co"># AqNPATisMix outline=T for LBO</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html">boxplot</a></span>(<span class="kw">featureCount_repl1</span>,
  add = <span class="kw">F</span>,
  ylim = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="op">-</span><span class="fl">4</span>, <span class="fl">4</span>), 
  staplewex = <span class="fl">0</span>, <span class="co"># remove horizontal whisker lines</span>
  staplecol = <span class="st">"white"</span>, <span class="co"># just to be totally sure</span>
  outline = <span class="kw">F</span>, <span class="co"># remove outlying points</span>
  whisklty = <span class="fl">0</span>, <span class="co"># remove vertical whisker lines</span>
  las = <span class="fl">2</span>, horizontal = <span class="kw">F</span>, xaxt = <span class="st">"n"</span>,
  border = <span class="kw">pamCol_vec</span>[<span class="kw">sampDesc_repl1</span><span class="op">$</span><span class="kw">pam50_subtype</span>])
<span class="fu"><a href="https://rdrr.io/r/graphics/title.html">title</a></span>(<span class="st">"Coverage for repl1 samples ordered by bioSample ID"</span>)

<span class="fu"><a href="https://rdrr.io/r/graphics/legend.html">legend</a></span>(<span class="st">'top'</span>, text.col=<span class="kw">pamCol_vec</span>, legend=<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="kw">pamCol_vec</span>), bty=<span class="st">'n'</span>, horiz=<span class="kw">T</span>)
</pre></div>
<p><img src="reference/figures/README-bxp-repl1-1.png" width="100%"></p>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="co">###, fig.cap="Coverage densities for repl1 samples"}</span>

<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(mfcol = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>, <span class="fl">1</span>), mar = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">1</span>), oma = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>, <span class="fl">3</span>, <span class="fl">2</span>, <span class="fl">0</span>))


<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span>(<span class="fu"><a href="https://rdrr.io/r/stats/density.html">density</a></span>(<span class="kw">featureCount_repl1</span>[, <span class="fl">1</span>]),
  col = <span class="kw">pamCol_vec</span>[<span class="kw">sampDesc_repl1</span><span class="op">$</span><span class="kw">pam50_subtype</span>[<span class="fl">1</span>]],
  lty = <span class="fl">1</span>, lwd = <span class="fl">2</span>, 
  xlim = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="op">-</span><span class="fl">4</span>, <span class="fl">7</span>), ylim = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>, <span class="fl">.4</span>), las = <span class="fl">2</span>, main = <span class="st">""</span>, xlab = <span class="st">""</span>
)
<span class="fu"><a href="https://rdrr.io/r/graphics/title.html">title</a></span>(<span class="st">"Coverage densities for repl1 samples"</span>)

<span class="co">for</span> (<span class="kw">JJ</span> <span class="co">in</span> <span class="fl">2</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span>(<span class="kw">featureCount_repl1</span>)) {
  <span class="kw">den</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/density.html">density</a></span>(<span class="kw">featureCount_repl1</span>[, <span class="kw">JJ</span>])
  <span class="fu"><a href="https://rdrr.io/r/graphics/lines.html">lines</a></span>(<span class="kw">den</span><span class="op">$</span><span class="kw">x</span>, <span class="kw">den</span><span class="op">$</span><span class="kw">y</span>,
    col = <span class="kw">pamCol_vec</span>[<span class="kw">sampDesc_repl1</span><span class="op">$</span><span class="kw">pam50_subtype</span>[<span class="kw">JJ</span>]],
    lty = <span class="fl">1</span>)
}

<span class="fu"><a href="https://rdrr.io/r/graphics/legend.html">legend</a></span>(<span class="st">'top'</span>, text.col=<span class="kw">pamCol_vec</span>, legend=<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="kw">pamCol_vec</span>), bty=<span class="st">'n'</span>, horiz=<span class="kw">T</span>)
</pre></div>
<p><img src="reference/figures/README-density-repl1-1.png" width="100%"></p>
<p>FPKM of zero looks like a good place to separate expressed from weakly expressed genes. Apply this filter, requiring at a gene to be epxressed beyond this nominal value in at least 10 samples to be considered for this analysis.</p>
<div class="sourceCode" id="cb11"><pre class="downlit">
   <span class="co">### , fig.cap="Filtered Coverage for repl1 samples orderd bi bioSample ID"}</span>

<span class="kw">weak.flg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span>(<span class="kw">featureCount_repl1</span> <span class="op">&gt;</span> <span class="fl">0</span>)  <span class="op">&lt;</span> <span class="fl">10</span>

<span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span>(<span class="st">"Excluding"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span>(<span class="fl">100</span><span class="op">*</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span>(<span class="kw">weak.flg</span>),<span class="fl">1</span>), <span class="st">"percent of genes.\n"</span>)
<span class="co">#&gt; Excluding 44.5 percent of genes.</span>
<span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="op">!</span><span class="kw">weak.flg</span>), <span class="st">'genes kept for analysis.\n'</span>)
<span class="co">#&gt; 17115 genes kept for analysis.</span>

<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(mfrow = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>, <span class="fl">1</span>), mar = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">4</span>), oma = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">4</span>, <span class="fl">2</span>))

<span class="co"># AqNPATisMix outline=T for LBO</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html">boxplot</a></span>(<span class="kw">featureCount_repl1</span>[<span class="op">!</span><span class="kw">weak.flg</span>,],
  add = <span class="kw">F</span>,
  ylim = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="op">-</span><span class="fl">1</span>, <span class="fl">5</span>), 
  staplewex = <span class="fl">0</span>, <span class="co"># remove horizontal whisker lines</span>
  staplecol = <span class="st">"white"</span>, <span class="co"># just to be totally sure</span>
  outline = <span class="kw">F</span>, <span class="co"># remove outlying points</span>
  whisklty = <span class="fl">0</span>, <span class="co"># remove vertical whisker lines</span>
  las = <span class="fl">2</span>, horizontal = <span class="kw">F</span>, xaxt = <span class="st">"n"</span>,
  border = <span class="kw">pamCol_vec</span>[<span class="kw">sampDesc_repl1</span><span class="op">$</span><span class="kw">pam50_subtype</span>])
<span class="fu"><a href="https://rdrr.io/r/graphics/title.html">title</a></span>(<span class="st">"Filtered Coverage for repl1 samples ordered by bioSample ID"</span>)

<span class="fu"><a href="https://rdrr.io/r/graphics/legend.html">legend</a></span>(<span class="st">'top'</span>, text.col=<span class="kw">pamCol_vec</span>, legend=<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="kw">pamCol_vec</span>), bty=<span class="st">'n'</span>, horiz=<span class="kw">T</span>)
</pre></div>
<p><img src="reference/figures/README-bxp-repl1-filtered-1.png" width="100%"></p>
<div class="sourceCode" id="cb12"><pre class="downlit">
   <span class="co">### , fig.cap='MDS plots filtered coverage in repl1 samples', echo=T}</span>

<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(mfcol = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>, <span class="fl">2</span>), mar = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">4</span>, <span class="fl">4</span>, <span class="fl">2</span>, <span class="fl">1</span>), xpd = <span class="fl">NA</span>, oma = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">2</span>, <span class="fl">0</span>))

<span class="kw">MDS.out</span> <span class="op">&lt;-</span> <span class="kw">limma</span>::<span class="fu"><a href="https://rdrr.io/pkg/limma/man/plotMDS.html">plotMDS</a></span>(<span class="kw">featureCount_repl1</span>[<span class="op">!</span><span class="kw">weak.flg</span>,],
  col = <span class="kw">pamCol_vec</span>[<span class="kw">sampDesc_repl1</span><span class="op">$</span><span class="kw">pam50_subtype</span>],
  pch=<span class="fl">1</span>
)


<span class="kw">MDS.out</span> <span class="op">&lt;-</span> <span class="kw">limma</span>::<span class="fu"><a href="https://rdrr.io/pkg/limma/man/plotMDS.html">plotMDS</a></span>(<span class="kw">featureCount_repl1</span>[<span class="op">!</span><span class="kw">weak.flg</span>,],
  col = <span class="kw">pamCol_vec</span>[<span class="kw">sampDesc_repl1</span><span class="op">$</span><span class="kw">pam50_subtype</span>],
  pch=<span class="fl">1</span>,
  dim.plot = <span class="fl">3</span><span class="op">:</span><span class="fl">4</span>
)

<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span>(outer=<span class="kw">T</span>, side=<span class="fl">3</span>,
 <span class="st">"MDS plots filtered coverage in repl1 samples - Col = pam50_subtype"</span>)
</pre></div>
<p><img src="reference/figures/README-plotMDS-repl1-1.png" width="100%"></p>
<p>We see some association with pam subtype, but no strong clustering in this display.</p>
</div>
</div>
<div id="package-page" class="section level2">
<h2 class="hasAnchor">
<a href="#package-page" class="anchor"></a>Package page</h2>
<ul>
<li><a href="https://12379monty.github.io/GSE96058/" class="uri">https://12379monty.github.io/GSE96058/</a></li>
</ul>
<!-- DOESNT WORK WITH GITHUB
## References
<div id="refs"></div>
--><hr>
<p>Please note that the ‘GSE96058’ project is released with a <a href="CODE_OF_CONDUCT.html">Contributor Code of Conduct</a>. By contributing to this project, you agree to abide by its terms.</p>
<hr>
</div>
<div id="references" class="section level2">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<div id="refs" class="references">
<div id="ref-Brueffer:2018aa">
<p>1. Brueffer, C., Vallon-Christersson, J., Grabau, D., Ehinger, A., Häkkinen, J., Hegardt, C., Malina, J., Chen, Y., Bendahl, P.-O., and Manjer, J. <em>et al.</em> Clinical value of rna sequencing-based classifiers for prediction of the five conventional breast cancer biomarkers: A report from the population-based multicenter sweden cancerome analysis network-breast initiative. JCO precision oncology <em>2</em>, PO.17.00135. Available at: <a href="https://pubmed.ncbi.nlm.nih.gov/32913985" class="uri">https://pubmed.ncbi.nlm.nih.gov/32913985</a>.</p>
</div>
<div id="ref-Varley:2014aa">
<p>2. Varley, K.E., Gertz, J., Roberts, B.S., Davis, N.S., Bowling, K.M., Kirby, M.K., Nesmith, A.S., Oliver, P.G., Grizzle, W.E., and Forero, A. <em>et al.</em> (2014). Recurrent read-through fusion transcripts in breast cancer. Breast Cancer Research and Treatment <em>146</em>, 287–297. Available at: <a href="https://doi.org/10.1007/s10549-014-3019-2" class="uri">https://doi.org/10.1007/s10549-014-3019-2</a>.</p>
</div>
<div id="ref-Li:2018aa">
<p>3. Li, X., Rouchka, E.C., Brock, G.N., Yan, J., O’Toole, T.E., Tieri, D.A., and Cooper, N.G.F. A combined approach with gene-wise normalization improves the analysis of rna-seq data in human breast cancer subtypes. PloS one <em>13</em>, e0201813–e0201813. Available at: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30089167" class="uri">https://www.ncbi.nlm.nih.gov/pubmed/30089167</a>.</p>
</div>
</div>
<!--

Rscript -e "rmarkdown::render('README.Rmd')"

# nohup Rscript -e "rmarkdown::render('README.Rmd')" > README.log  &


-->
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Francois Collin <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-6202-0141" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Francois Collin.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
